generate/templates/partials/async_function.cc

```diff
@@ -56,20 +56,21 @@ NAN_METHOD({{ cppClassName }}::{{ cppFunctionName }}) {
         {% if arg.cppClassName == 'Array' %}
           {
             v8::Local<v8::Array> tempArray = v8::Local<v8::Array>::Cast(info[{{ arg.jsArg }}]);
-            baton->{{ arg.name }} = new {{ arg.cType|unPointer }}[tempArray->Length()];
+            baton->{{ arg.name }} = ({{ arg.cType|unPointer }}*)malloc(sizeof({{ arg.cType|unPointer }}) * tempArray->Length());
             for (uint32_t i = 0; i < tempArray->Length(); ++i) {
               auto conversionResult = Configurable{{ arg.arrayElementCppClassName }}::fromJavascript(
                 nodegitContext,
                 Nan::Get(tempArray, i).ToLocalChecked()
               );
 
               if (!conversionResult.result) {
-                delete[] baton->{{ arg.name }};
+                // TODO free previously allocated memory
```

Comment:
Does this still need to be handled?

Reply:
@julianmesa-gitkraken this commit came from you.

Reply:
Yes but I will do in another PR. I need to store all allocated memory in a map, then if I get an error reading javascript arguments we have to release all allocated memory, then throw the error.